sycl/doc/extensions/DeviceGlobal/cl_intel_global_variable_access.asciidoc

```diff
@@ -0,0 +1,265 @@
+device has a distinct instance of any global variables contained by that
+program object.  These functions read or write the variable instance that
+resides on the target device of the _command_queue_.
+
+These functions are non-blocking, so they may return even before data is read
+or written to the global variable.  The _event_ argument returns an event
+object which the application can use to query the execution status of the
+command.  The application must not use the content of the _ptr_ buffer until
+the command has completed, and the value read by
+*clEnqueueReadGlobalVariableINTEL* is not available in the buffer referenced by
+_ptr_ until after the command has completed.
+
+*clEnqueueReadGlobalVariableINTEL* and *clEnqueueWriteGlobalVariableINTEL*
+return `CL_SUCCESS` if the function is executed successfully.  Otherwise, they
+return one of the following errors:
+
+* `CL_INVALID_COMMAND_QUEUE` if _command_queue_ is not a valid host
+  command-queue.
+
+* `CL_INVALID_CONTEXT` if the context associated with _command_queue_ and
+  _program_ are not the same or if the context associated with _command_queue_
+  and events in _event_wait_list_ are not the same.
+
+* `CL_INVALID_PROGRAM` if _program_ is not a valid program object.
+
+* `CL_INVALID_PROGRAM_EXECUTABLE` if there is no successfully built program
+  executable available for device associated with _command_queue_.
+
+* `CL_INVALID_ARG_VALUE` if _name_ does not identify a global variable defined
+  in _program_.
+
+* `CL_INVALID_VALUE` if _name_ or _ptr_ are NULL.
+
+* `CL_INVALID_VALUE` if the region being read or written specified by
+  (_offset_, _size_) is not fully contained by the size of the global variable.
+
+* `CL_INVALID_EVENT_WAIT_LIST` if _event_wait_list_ is NULL and
+  _num_events_in_wait_list_ > 0, or _event_wait_list_ is not NULL and
+  _num_events_in_wait_list_ is 0, or if event objects in _event_wait_list_ are
+  not valid events.
+
+* `CL_INVALID_OPERATION` if *clEnqueueReadGlobalVariableINTEL* is called for a
+  global variable that is not readable from the host.  This is the case when
+  _program_ is created from a SPIR-V module that declares the
+  *GlobalVariableDecorationsINTEL* capability and the *OpVariable* is decorated
+  with *HostAccessINTEL* and _Write_ or _None_ access mode.
+
+* `CL_INVALID_OPERATION` if *clEnqueueWriteGlobalVariableINTEL* is called for a
+  global variable that is not writable from the host.  This is the case when
+  _program_ is created from a SPIR-V module that declares the
+  *GlobalVariableDecorationsINTEL* capability and the *OpVariable* is decorated
+  with *HostAccessINTEL* and _Read_ or _None_ access mode.
+
+* `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required by
+  the OpenCL implementation on the device.
+
+* `CL_OUT_OF_HOST_MEMORY` if there is a failure to allocate resources required
+  by the OpenCL implementation on the host.
+
```

Comment:
Need to add another error code to support issues with the new blocking parameter

CL_EXEC_STATUS_ERROR_FOR_EVENTS_IN_WAIT_LIST if the copy operation is blocking and the execution status of any of the events in event_wait_list is a negative integer value.

Reply:
Done in 05d1c30f0b9247c728be8a50ba6c7a2c57e94628.